Source/FullScreenPokemon.d.ts

```diff
@@ -1916,6 +1921,11 @@ declare module FullScreenPokemon {
          * The partyActivate Function used to interact with this HMCharacter.
          */
         moveCallback: (player: IPlayer, pokemon: IPokemon) => void;
+
+        /**
+         * Indicates if the Player is entering the water.
+         */
+        enteringWater?: boolean;
```

Comment:
This shouldn't be on IHmCharacter; it should be specific to the WaterEdge objects.